Doc/library/pathlib.rst

```diff
@@ -1624,8 +1624,7 @@ example because the path doesn't exist).
    Remove this file or symbolic link.  If the path points to a directory,
    use :func:`Path.rmdir` instead.
 
-   If *missing_ok* is false (the default), :exc:`FileNotFoundError` is
-   raised if the path does not exist.
+   This method propagates any :exc:`OSError` encountered during removal.
```

Comment:
```suggestion
   If *missing_ok* is false (the default), this method propagates any 
   :exc:`OSError` encountered during removal.
```

Reply:
Hi @barneygale - is that technically accurate. My understanding is that any `OSError` is propagated regardless of `missing_ok`, which suppresses any `FileNotFoundError`.
I understand the interest to specifically reference the `FileNotFoundError` in the context of `missing_ok` ... how about this new suggestion?

Reply:
```
   This method propagates any :exc:`OSError` encountered during removal. 
    For example: If *missing_ok* is false (the default), :exc:`FileNotFoundError` is 
     raised if the path does not exist. 
 
 
     If *missing_ok* is true, :exc:`FileNotFoundError` exceptions will be ...
```

Reply:
`FileNotFoundError` is a subclass of `OSError`! Maybe we should make that clearer:

```suggestion
   If *missing_ok* is false (the default), this method propagates any 
   :exc:`OSError` from the operating system, including :exc:`FileNotFoundError`.
```

Reply:
Thanks, I've updated with that text and slightly extended the following paragraph:

```rst
   If *missing_ok* is false (the default), this method propagates any 
   :exc:`OSError` from the operating system, including :exc:`FileNotFoundError`.

   If *missing_ok* is true, :exc:`FileNotFoundError` exceptions will be
   ignored (same behavior as the POSIX ``rm -f`` command), any other
   :exc:`OSError` which is encountered will continue to be propogated.
```